logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/ValidateAssetMimeTypeUseCase.kt

```diff
@@ -18,19 +18,22 @@
 package com.wire.kalium.logic.feature.asset
 
 /**
- * Returns true if the mime type is allowed and false otherwise.
- * @param mimeType the mime type to validate.
+ * Returns true if the file extension is present in file name and is allowed and false otherwise.
+ * @param fileName the file name (with extension) to validate.
  * @param allowedExtension the list of allowed extension.
  */
 interface ValidateAssetMimeTypeUseCase {
-    operator fun invoke(mimeType: String, allowedExtension: List<String>): Boolean
+    operator fun invoke(fileName: String, allowedExtension: List<String>): Boolean
 }
 
 internal class ValidateAssetMimeTypeUseCaseImpl : ValidateAssetMimeTypeUseCase {
-    override operator fun invoke(mimeType: String, allowedExtension: List<String>): Boolean {
-        val extension = mimeType.split("/").last().lowercase()
-        return allowedExtension.any {
-            it.lowercase() == extension
+    override operator fun invoke(fileName: String, allowedExtension: List<String>): Boolean {
```

Comment:
Question: What if we have multiple extensions hiding in the file or other type of tricks ? Would be good idea to follow some of these tips ? 

https://book.hacktricks.xyz/pentesting-web/file-upload#bypass-file-extensions-checks